### PR TITLE
Tests config fixes

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###

[[AB#591](https://dev.azure.com/hmcts/90472b63-dc8d-4ae6-8819-a32bd211e914/_workitems/edit/591)]

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -88,6 +88,48 @@ postgresql:
...
+## Smoke and functional tests
+
+From version 2.14.0 of this chart you can configure your functional and smoke tests to run just after deployment or at scheduled times 
+as cron jobs.
+
+```yaml
+java:
+  testsConfig:
+    keyVaults:
+      "cmc":
```
Comment (timja) [inline README.md]: quotes aren't neneded
```suggestion
      cmc:
```

File: README.md [hunk 2]
```diff
@@ -88,6 +88,48 @@ postgresql:
...
+    image: hmctspublic.azurecr.io/spring-boot/template-test
+    enabled: true
+    environment:
+      TEST_URL: http://plum-recipe-backend-java-overridden
+
+  functionaltests:
+    image: hmctspublic.azurecr.io/spring-boot/template-test
+    enabled: true
+
+  smoketestscron:
```
Comment (timja) [inline README.md]: cron is a separate word, shouldn't we camel or kebab case?
```suggestion
  smoketestsCron:
```

```suggestion
  smoketests-cron:
```

File: java/templates/_helpers.tpl [hunk 3]
```diff
@@ -89,15 +89,19 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- end -}}
 
 {{- define "java.tests.metadata" -}}
```
Comment (timja) [inline java/templates/_helpers.tpl]: version bump required? there could be released charts in the wild with this?

File: README.md (hunk 2 above)
Comment (luigibk) [reply README.md]: Camel causes trouble with dns naming convetions, so it's trouble. Kebab case would be possible, but I like it better in one word to be honest

File: java/templates/_helpers.tpl (hunk 3 above)
Comment (luigibk) [reply java/templates/_helpers.tpl]: Bumping to 2.14.0 ... but I am not sure there are charts released with "java.tests.metadata"

Comment (timja) [reply java/templates/_helpers.tpl]: i mean these are global namespaced, if someone imports another chart that  has the old version you could get that version of it the helper

**Pull request merged**